xtext-gradle-plugin/src/main/java/org/xtext/gradle/tasks/XtextGenerate.xtend

```diff
@@ -124,23 +130,29 @@ class XtextGenerate extends DefaultTask {
 		]
 	}
 
-	private def addIncrementalInputs(GradleBuildRequest request, IncrementalTaskInputs inputs) {
+	private def addIncrementalInputs(GradleBuildRequest request, InputChanges inputs) {
 		request.incremental = options.incremental && inputs.incremental
 
-		inputs.outOfDate[
-			if (allSources.contains(file)) {
-				request.dirtyFiles += file
-			}
-			if (getNullSafeClasspath.contains(file)) {
-				request.dirtyClasspathEntries += file
+		for (change : inputs.getFileChanges(allSources)) {
+			if (change.changeType == ChangeType.MODIFIED || change.changeType == ChangeType.ADDED) {
+				this.logger.error("mimimi1 " + change.file)
+				request.dirtyFiles += change.file
+			} else if (change.changeType == ChangeType.REMOVED) {
+				request.deletedFiles += change.file
 			}
-		]
-
-		inputs.removed[
-			if (allSources.contains(file)) {
-				request.deletedFiles += file
+		}
+		
+		val cp = getClasspath
+		if (cp !== null) {
+			for (change : inputs.getFileChanges(cp)) {
+				if (cp.contains(change.file)) {
+					if (change.changeType == ChangeType.MODIFIED || change.changeType == ChangeType.ADDED) {
```

Comment:
should we also check for (deleted) jar on cp here?

Reply:
Yes, absolutely.